x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/zero_shot_inference_configuration.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import { EuiComboBox, EuiFormRow, EuiSpacer } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { IndexViewLogic } from '../../index_view_logic';
+
+import { MLInferenceLogic } from './ml_inference_logic';
+
+type LabelOptions = Array<{ label: string }>;
+
+export const ZeroShotClassificationInferenceConfiguration: React.FC = () => {
+  const { ingestionMethod } = useValues(IndexViewLogic);
+  const {
+    addInferencePipelineModal: { configuration },
+  } = useValues(MLInferenceLogic);
+  const { setInferencePipelineConfiguration } = useActions(MLInferenceLogic);
+
+  const zeroShotLabels = configuration?.inferenceConfig?.zero_shot_classification?.labels ?? [];
+  const labelOptions = zeroShotLabels.map((label) => ({ label }));
+
+  const onLabelChange = (selectedLabels: LabelOptions) => {
+    const inferenceConfig =
+      selectedLabels.length === 0
+        ? undefined
+        : {
+            zero_shot_classification: {
+              ...(configuration?.inferenceConfig?.zero_shot_classification ?? {}),
+              labels: selectedLabels.map(({ label }) => label),
+            },
+          };
+    setInferencePipelineConfiguration({
+      ...configuration,
+      inferenceConfig,
+    });
+  };
+  const onCreateLabel = (labelValue: string, labels: LabelOptions = []) => {
+    const normalizedLabelValue = labelValue.trim().toLowerCase();
+    if (!normalizedLabelValue) return;
+
+    const existingLabel = labels.find((label) => label.label === labelValue.trim());
```

Comment:
I assume we want to use `normalizedLabelValue` here instead of `labelValue.trim()`?

Reply:
I think I can get rid of the `toLowerCase` and then use it yes.